### PR TITLE
Centos7

### PR DESCRIPTION
**Pull request by jjasghar**
This _should_ give CentOS7 support.

### CODE REVIEW

**Comment by PierreRambaud** [comment]
:+1:

**Comment by jjasghar** [comment]
It looks like multi-nova is having an issue:
```ruby
  ================================================================================
                   Error executing action `start` on resource 'service[nova-compute]'
                   ================================================================================
 
                   Mixlib::ShellOut::ShellCommandFailed
                   ------------------------------------
                   Expected process to exit with [0], but received '1'
                   ---- Begin output of /bin/systemctl start openstack-nova-compute ----
                   STDOUT:
                   STDERR: Job for openstack-nova-compute.service failed. See 'systemctl status openstack-nova-compute.service' and 'journalctl -xn' for details.
                   ---- End output of /bin/systemctl start openstack-nova-compute ----
                   Ran /bin/systemctl start openstack-nova-compute returned 1
 
                   Resource Declaration:
                   ---------------------
                   # In /var/chef/cache/cookbooks/openstack-compute/recipes/compute.rb
 
                    79: service 'nova-compute' do
                    80:   service_name platform_options['compute_compute_service']
                    81:   supports status: true, restart: true
                    82:   subscribes :restart, resources('template[/etc/nova/nova.conf]')
                    83:
                    84:   action [:enable, :start]
                    85: end
 
                   Compiled Resource:
                   ------------------
                   # Declared in /var/chef/cache/cookbooks/openstack-compute/recipes/compute.rb:79:in `from_file'
 
                   service("nova-compute") do
                     action [:enable, :start]
                     updated true
                     supports {:status=>true, :restart=>true}
                     retries 0
                     retry_delay 2
                     default_guard_interpreter :default
                     service_name "openstack-nova-compute"
                     enabled true
                     pattern "nova-compute"
                     declared_type :service
                     cookbook_name "openstack-compute"
                     recipe_name "compute"
                   end
 
               [2015-02-19T15:06:48+00:00] INFO: Running queued delayed notifications before re-raising exception
               [2015-02-19T15:06:48+00:00] INFO: template[/etc/nova/nova.conf] sending restart action to service[nova-network] (delayed)
               Recipe: openstack-compute::network
                 * service[nova-network] action restart[2015-02-19T15:06:48+00:00] INFO: Processing service[nova-network] action restart (openstack-compute::network line 40)

```

**Comment by PierreRambaud** [comment]
HI,

For me I was having this error:
```
                
                     ================================================================================
                     Error executing action `start` on resource 'service[nova-api-metadata]'
                     ================================================================================
                     
                     Mixlib::ShellOut::ShellCommandFailed
                     ------------------------------------
                     Expected process to exit with [0], but received '1'
                     ---- Begin output of /bin/systemctl start openstack-nova-metadata-api ----
                     STDOUT: 
                     STDERR: Job for openstack-nova-metadata-api.service failed. See 'systemctl status openstack-nova-metadata-api.service' and 'journalctl -xn' for details.
                     ---- End output of /bin/systemctl start openstack-nova-metadata-api ----
                     Ran /bin/systemctl start openstack-nova-metadata-api returned 1
                     
                     Resource Declaration:
                     ---------------------
                     # In /var/chef/cache/cookbooks/openstack-compute/recipes/api-metadata.rb
                     
                      45: service 'nova-api-metadata' do
                      46:   service_name platform_options['compute_api_metadata_service']
                      47:   supports status: true, restart: true
                      48:   subscribes :restart, resources('template[/etc/nova/nova.conf]')
                      49: 
                      50:   action [:enable, :start]
                      51: end
                      52: 
                     
                     Compiled Resource:
                     ------------------
                     # Declared in /var/chef/cache/cookbooks/openstack-compute/recipes/api-metadata.rb:45:in `from_file'
                     
                     service("nova-api-metadata") do
                       action [:enable, :start]
                       supports {:status=>true, :restart=>true}
                       retries 0
                       retry_delay 2
                       default_guard_interpreter :default
                       service_name "openstack-nova-metadata-api"
                       enabled true
                       pattern "nova-api-metadata"
                       declared_type :service
                       cookbook_name "openstack-compute"
                       recipe_name "api-metadata"
                     end
```

And getting this python error:
```
2015-02-20 12:50:59.689 6336 ERROR nova.wsgi [-] Could not bind to 0.0.0.0:8775
2015-02-20 12:50:59.690 6336 CRITICAL nova [-] error: [Errno 98] Address already in use
2015-02-20 12:50:59.690 6336 TRACE nova Traceback (most recent call last):
2015-02-20 12:50:59.690 6336 TRACE nova   File "/usr/bin/nova-api-metadata", line 10, in <module>
2015-02-20 12:50:59.690 6336 TRACE nova     sys.exit(main())
2015-02-20 12:50:59.690 6336 TRACE nova   File "/usr/lib/python2.7/site-packages/nova/cmd/api_metadata.py", line 52, in main
2015-02-20 12:50:59.690 6336 TRACE nova     server = service.WSGIService('metadata', use_ssl=should_use_ssl)
2015-02-20 12:50:59.690 6336 TRACE nova   File "/usr/lib/python2.7/site-packages/nova/service.py", line 351, in __init__
2015-02-20 12:50:59.690 6336 TRACE nova     max_url_len=max_url_len)
2015-02-20 12:50:59.690 6336 TRACE nova   File "/usr/lib/python2.7/site-packages/nova/wsgi.py", line 131, in __init__
2015-02-20 12:50:59.690 6336 TRACE nova     self._socket = eventlet.listen(bind_addr, family, backlog=backlog)
2015-02-20 12:50:59.690 6336 TRACE nova   File "/usr/lib/python2.7/site-packages/eventlet/convenience.py", line 39, in listen
2015-02-20 12:50:59.690 6336 TRACE nova     sock.bind(addr)
2015-02-20 12:50:59.690 6336 TRACE nova   File "/usr/lib64/python2.7/socket.py", line 224, in meth
2015-02-20 12:50:59.690 6336 TRACE nova     return getattr(self._sock,name)(*args)
2015-02-20 12:50:59.690 6336 TRACE nova error: [Errno 98] Address already in use
```

I fixed it by adding this line in `vagrant-multi-centos7-nova.json` in `compute` key.
`"enabled_apis": "ec2,osapi_compute",`

With this, the metada api server can be started individually.

Regards

**Comment by cmluciano** [comment]
I'm generally +1 on this. I can get aio_nova to work, and I had multi_nova stood up and one point but I kept getting 500s with glance and launching images. 

@kramvan1 and I discovered that something seems to restart MySQL and 30-45 seconds later I can normally use the system as expected. This seems to happen with plain aio_nova too so I will go back and try to converge multi_nova again today.

**Comment by cmluciano** [comment]
Ok so I get something similar to @PierreRambaud 

```
    - restart service service[nova-api-os-compute]
                 [2015-02-20T16:58:45+00:00] INFO: template[/etc/nova/nova.conf] sending restart action to service[nova-api-metadata] (delayed)
                 Recipe: openstack-compute::api-metadata
                   * service[nova-api-metadata] action restart[2015-02-20T16:58:45+00:00] INFO: Processing service[nova-api-metadata] action restart (openstack-compute::api-metadata line 45)


                     ================================================================================
                     Error executing action `restart` on resource 'service[nova-api-metadata]'
                     ================================================================================

                     Mixlib::ShellOut::ShellCommandFailed
                     ------------------------------------
                     Expected process to exit with [0], but received '1'
                     ---- Begin output of /bin/systemctl restart openstack-nova-metadata-api ----
                     STDOUT:
                     STDERR: Job for openstack-nova-metadata-api.service failed. See 'systemctl status openstack-nova-metadata-api.service' and 'journalctl -xn' for details.
                     ---- End output of /bin/systemctl restart openstack-nova-metadata-api ----
                     Ran /bin/systemctl restart openstack-nova-metadata-api returned 1

                     Resource Declaration:
                     ---------------------
                     # In /var/chef/cache/cookbooks/openstack-compute/recipes/api-metadata.rb

                      45: service 'nova-api-metadata' do
                      46:   service_name platform_options['compute_api_metadata_service']
                      47:   supports status: true, restart: true
                      48:   subscribes :restart, resources('template[/etc/nova/nova.conf]')
                      49:
                      50:   action [:enable, :start]
                      51: end
                      52:

                     Compiled Resource:
                     ------------------
                     # Declared in /var/chef/cache/cookbooks/openstack-compute/recipes/api-metadata.rb:45:in `from_file'

                     service("nova-api-metadata") do
                       action [:enable, :start]
                       updated true
                       supports {:status=>true, :restart=>true}
                       retries 0
                       retry_delay 2
                       default_guard_interpreter :default
                       service_name "openstack-nova-metadata-api"
                       enabled true
                       pattern "nova-api-metadata"
                       declared_type :service
                       cookbook_name "openstack-compute"
```

File: environments/vagrant-multi-centos7-nova.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            "compute": {
+                "libvirt": {
+                    "virt_type": "qemu"
+                },
+                "network": {
+                    "public_interface": "enp0s3",
+                    "service_type": "nova",
+                    "multi_host": "true"
+                },
+                "enabled_apis": "ec2,osapi_compute",
```
Comment (jjasghar) [inline environments/vagrant-multi-centos7-nova.json]: yep, it's up now :)

**Pull request merged**